### ISSUE TITLE
How to add user attributes to trials in huggingface trainer

### ISSUE THREAD

**Issue by fancyerii** [optuna/optuna#6410]
### Don't use GitHub Issues to ask support questions.

I am using optuna with huggingface trainer. Here is my code:

```
    def optuna_hp_space(trial):
        return {
            "learning_rate": trial.suggest_float("learning_rate", 0.5e-6, 0.5e-4, log=True),
            "weight_decay": trial.suggest_float("weight_decay", 0.0, 0.3),
        }

        def compute_objective(metrics):
            return metrics["f1"]

        best_run = trainer.hyperparameter_search(
            direction="maximize",
            backend="optuna",
            hp_space=optuna_hp_space,
            n_trials=args.n_trials,
            compute_objective=compute_objective,
            study_name=args.study_name,
            storage="sqlite:///optuna_trials.db",
            load_if_exists=True,
            pruner=MedianPruner(n_startup_trials=5, n_warmup_steps=2),
        )
```
Besides the objective "f1", I also want to see other metrics such as accuracy in the dashboard. Acorrding to https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/003_attributes.html#adding-user-attributes-to-trials, It's very easy to add User Attributes to Trials by:

```
trial.set_user_attr("accuracy", accuracy)
```

But in trainer, I can't get trail object. I asked this question to gemini, it says I can get trial by:

```
trial = optuna.get_current_trial()
```
But when I ran, it raise exception.
Now I am hacking this problem by using global variable to store trial object in optuna_hp_space function and use it in compute_objective. But if in multithreads environment, I don't know whether it works.

```
_active_trial = None

    def optuna_hp_space(trial):
        global _active_trial
        _active_trial = trial


     def compute_objective(metrics):
         _active_trial.set_user_attr(key, value)
                    
```

my environment:
```
optuna                    4.6.0
accelerate                1.12.0 
transformers              4.57.3
```
